pkg/metrics/metrics.go

```diff
@@ -69,14 +80,36 @@ func NewPrometheusNodeConfig(clientset *kubernetes.Clientset) (*PrometheusNodeCo
 	}, err
 }
 
+// newMetricsConfig creates a new instance for metricsConfig  to be used by the caller.
+func newMetricsConfig(cfg *rest.Config) (*metricsConfig, error) {
+	if cfg == nil {
+		return nil, errors.New("could not get config for creating metrics configuration")
+	}
+	oclient, err := k8sclient.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create config clientset")
+	}
+	mclient, err := monclient.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create monitoring clientset")
+	}
+	return &metricsConfig{oclient, mclient}, nil
+}
+
 // Validate will verify if cluster monitoring is enabled in the operator namespace
+// and delete any stale metrics resources from previous operator version
 func Validate(ctx context.Context, cfg *rest.Config, namespace string) error {
-	oclient, err := k8sclient.NewForConfig(cfg)
+	metricsConfig, err := newMetricsConfig(cfg)
 	if err != nil {
-		return errors.Wrap(err, "could not create config clientset")
+		return errors.Wrap(err, "unable to initialize metrics config")
+	}
+
+	// Remove any existing stale resources from previous operator version.
+	if err := metricsConfig.removeStaleResources(ctx, cfg, namespace); err != nil {
+		return errors.Wrap(err, "could not remove stale metrics resources")
```

Comment:
You should consider separation of concerns here. This shouldn't be part of validation.